game.py

```diff
@@ -20,7 +20,29 @@ def snowman(snowman_word):
     If the player wins and, 
     'Sorry, you lose! The word was {snowman_word}' if the player loses
     """
-    pass
+    letter_status = build_letter_status_dict(snowman_word)
+    wrong_guesses = []
+    wrong_count = 0
+    
+    while wrong_count < SNOWMAN_MAX_WRONG_GUESSES:
+        print_snowman_graphic(wrong_count)
+        print_word_progress_string(snowman_word, letter_status)
+        
+        if is_word_guessed(snowman_word, letter_status):
+            print('Congratulations, you win!')
+            return
+        
+        letter = get_letter_from_user(letter_status, wrong_guesses)
+        
+        if letter in snowman_word:
```

Comment:
When we receive a letter from the user, we want to check to see if it exists within the word or not. Given the way this program is set up, we end up with two different collections that hold all the letters of the word, a string (`snowman_word`) and a dictionary (`letter_status`). 

While we could check either one for the letter we've received, searching within a dictionary is ever so slightly more efficient than searching through a string (We'll talk more about why in Unit 1)! With that in mind, we could make the following small tweak to increase the efficiency of our solution:

```suggestion
        if letter in letter_status:
```

Reply:
thanks